### PR TITLE
chore: Extend Renovate group:kubernetes

### PR DESCRIPTION
**Pull request by marc1404**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:

This PR adapts the configuration for [Renovate](https://docs.renovatebot.com/) to group more dependencies related to Kubernetes updates than included by the default `group:kubernetes` preset.

* Since we extend `config:recommended` the preset `group:kubernetes` is included and needs to be ignored through `ignorePresets` if we want to use our own rules.
* The data sources are extended to include `github-releases` to group PRs such as this one: https://github.com/gardener/gardener/pull/11017
* The matched package names are extended to include `kubernetes/kubernetes` (for grouping https://github.com/gardener/gardener/pull/11017) and `k8s.io/component-helpers (for grouping https://github.com/gardener/gardener/pull/11019).

In short, this PR aims to group the following Renovate PRs:
* https://github.com/gardener/gardener/pull/11017
* https://github.com/gardener/gardener/pull/11018
* https://github.com/gardener/gardener/pull/11019

**Which issue(s) this PR fixes**:

_n.a._

**Special notes for your reviewer**:

/cc @oliver-goetz 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by oliver-goetz** [comment]
How about creating an upstream PR to update the [`kubernetes packages`](https://github.com/renovatebot/renovate/blob/10f1f46d382e83dbc92140186b2e4fbf972ea645/lib/config/presets/internal/group.ts#L301-L339) group instead? In this case we would not need to duplicate so much config.

It makes sense to add `'k8s.io/component-helpers'` to the preset, but adding `'kubernetes/kubernetes'` and `github-releases` would have no effect because it has version like `1.31.4` while the other packages use `0.31.4`.

**Comment by marc1404** [comment]
> How about creating an upstream PR to update the [`kubernetes packages`](https://github.com/renovatebot/renovate/blob/10f1f46d382e83dbc92140186b2e4fbf972ea645/lib/config/presets/internal/group.ts#L301-L339) group instead? In this case we would not need to duplicate so much config.
> 
> It makes sense to add `'k8s.io/component-helpers'` to the preset, but adding `'kubernetes/kubernetes'` and `github-releases` would have no effect because it has version like `1.31.4` while the other packages use `0.31.4`.

@oliver-goetz Oh 🤦‍♂️ Of course! I initially wanted to go the upstream contribution route to avoid duplication, but somehow, I missed the version mismatch between the Go dependencies and the `kubernetes/kubernetes` package.

Let's close this and I'll open a PR in the Renovate repository instead to extend the `group:kubernetes` 👍 
Thank you!

**Comment by marc1404** [comment]
ℹ️ https://github.com/renovatebot/renovate/pull/33223
